src/globus_cli/commands/flows/create.py

```diff
@@ -0,0 +1,196 @@
+from __future__ import annotations
+
+import typing as t
+
+import click
+
+from globus_cli.login_manager import LoginManager
+from globus_cli.parsing import JSONStringOrFile, command
+from globus_cli.termio import Field, TextMode, display, formatters
+
+ROLE_TYPES = ("flow_viewer", "flow_starter", "flow_administrator", "flow_owner")
+
+
+@command("create", short_help="Create a flow")
+@click.argument(
+    "title",
+    type=str,
+)
+@click.argument(
+    "definition",
+    type=JSONStringOrFile(),
+    metavar="DEFINITION",
+)
+@click.option(
+    "--input-schema",
+    "input_schema",
+    type=JSONStringOrFile(),
+    help="""
+        The JSON input schema that governs the parameters
+        used to start the flow.
+
+        The input document may be specified inline,
+        or it may be a path to a JSON file, prefixed with "file:".
+
+        Example: Inline JSON:
+
+        \b
+            --input_schema '{"properties": {"src": {"type": "string"}}}'
+
+        Example: Path to JSON file:
+
+        \b
+            --input_schema file:schema.json
+
+        If unspecified, the default is an empty JSON object ('{}').
+    """,
+)
+@click.option(
+    "--subtitle",
+    type=str,
+    help="A concise summary of the flow's purpose.",
+)
+@click.option(
+    "--description",
+    type=str,
+    help="A detailed description of the flow's purpose.",
+)
+@click.option(
+    "--administrator",
+    "administrators",
+    type=str,
+    multiple=True,
+    help="""
+        A principal that may perform administrative operations
+        on the flow (e.g., update, delete).
+
+        This option can be specified multiple times
+        to create a list of flow administrators.
+    """,
+)
+@click.option(
+    "--starter",
+    "starters",
+    type=str,
+    multiple=True,
+    help="""
+        A principal that may run the flow.
+
+        Use "all_authenticated_users" to make the flow visible to everyone.
+
+        This option can be specified multiple times
+        to create a list of flow starters.
+    """,
+)
+@click.option(
+    "--viewer",
+    "viewers",
+    type=str,
+    multiple=True,
+    help="""
+        A principal that may view the flow.
+
+        Use "public" to allow any authenticated user to run the flow.
+
+        This option can be specified multiple times
+        to create a list of flow viewers.
+    """,
+)
+@click.option(
+    "--keyword",
+    "keywords",
+    type=str,
+    multiple=True,
+    help="""
+        A term used to help discover this flow when
+        browsing and searching.
+
+        This option can be specified multiple times
+        to create a list of keywords.
+    """,
+)
```

Comment:
Rather than allowing `--keyword foo --keyword bar`, would a better CLI usage be `--keywords foo,bar`?
I think comma-delimited lists are nicer for most users, even if it forbids commas inside of keywords.

Reply:
I think that allowing CSVs would break with the existing idea that, for example, `--administrator` must be specified multiple times.

Reply:
I have a very soft lean toward multiple instances of the argument, but if there is substantial prior art for comma-delimited lists in the CLI then I'm willing to be persuaded that it's better to keep consistency at a CLI-level.

Reply:
We allow for both comma-delimited lists and multi-option usages. To summarize my views on the two modes of use in two points:

1. Multi-option use is very explicit, allows for more robust and complex parsing, but is also more onerous for users. Comma-delimited lists are very easy and lightweight, but more limited.
2. They're two equally good APIs for users, and should be tailored to specific use-cases, not one-size-fits-all.

GCS and Transfer both treat keywords as a comma-delimited lists, which is worth keeping on our radar. That's partly driven by a service behavior, in that Transfer actually requires keywords for endpoint search to be provided as a string field, with comma-delimited semantics service-side.

<details><summary>GCS tangent</summary>
GCS is a little bit inbetween, accepting a list of strings, but then needing to serialize that and send it to Transfer (to reflect the data there) as a comma-delimited list. So even though GCS might accept `["foo,bar", "baz"]`, it can't mean anything different from `"foo,bar,baz"`.
</details>

Whether or not cross-service consistency is a goal (I would vaguely like it but also don't think it's terribly realistic to expect it at present), I think the meaning of keywords for search and retrieval is better aligned with the more lightweight comma-delimited usage.
Regarding the other flavor of consistency -- that within a command or API -- I'm not sure that it makes sense to align the mode of use for `--keyword` with `--administrator` (to stick with that example) because they're two different kinds of data.

Reply:
> GCS and Transfer both treat keywords as a comma-delimited lists,

and

> I'm not sure that it makes sense to align the mode of use for `--keyword` with `--administrator` (to stick with that example) because they're two different kinds of data.

Flows accepts a list of strings, not a CSV. I don't currently think it's onerous to use multiple keywords.